### PR TITLE
docs(CONTRIBUTING): push beta releases to `release`

### PR DESCRIPTION
**Pull request by intelliot**
## High Level Overview of Change

Clarify that beta releases should also be pushed to the `release` branch.

### Context of Change

Beta releases are releases too. ([Example](https://github.com/XRPLF/rippled/commit/beba87129e6de43576b1f90a71370b16a081f40e))

Sections that were rewrapped were wrapped to 72 characters, the same as `rippled` commit messages.

### Type of Change

- [x] Documentation Updates

### CODE REVIEW

**Review by legleux — APPROVED** [review]
LGTM :+1:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,54 +1,51 @@
...
 When you submit a pull request, please link the corresponding XLS in the
-description.
-An XLS still in draft status is considered a work-in-progress and open for
-discussion.
-Please do not submit a pull request before allowing due time for questions,
-suggestions, and changes to the XLS draft.
-It is the responsibility of the XLS author to update the draft to match the
-final implementation when its corresponding pull request is merged.
+description. An XLS still in draft status is considered a
+work-in-progress and open for discussion. Please allow time time for
```
Comment (ckeshava) [inline CONTRIBUTING.md:44]: `time` is repeated twice in this line

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -79,10 +76,9 @@ Header includes must be [levelized](./Builds/levelization).
...
 
-Pull requests must target the `develop` branch.[^1]
+In general, pull requests use `develop` as the base branch.
 
-[^1]: There are exceptions to this policy for hotfixes, but no one consulting
-  this document will be in that situation.
+(Hotfixes are an exception.)
 
 Changes to pull requests must be added as new commits.
 You may **never force push a branch in a pull request** (e.g. after a rebase).
```
Comment (ckeshava) [inline CONTRIBUTING.md:84]: But is it possible to not-force-push a branch? In cases where we need to clean up the commit history, I feel force push is a last resort.

Comment (ckeshava) [reply CONTRIBUTING.md:84]: ok, these are suggestions. I read the below paragraph 👍

**Review by ckeshava — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (intelliot) [inline CONTRIBUTING.md:44]:
```suggestion
work-in-progress and open for discussion. Please allow time for
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (intelliot) [inline CONTRIBUTING.md:84]:
```suggestion
Once code reviewers have started looking at your code, please avoid
force-pushing a branch in a pull request.
```

**Pull request merged**